javascript/ql/src/semmle/javascript/frameworks/LazyCache.qll

```diff
@@ -0,0 +1,55 @@
+/**
+ * Models imports through the NPM `lazy-cache` package.
+ */
+
+import javascript
+
+module LazyCache {
+  /**
+   * A lazy-cache object, usually created through an expression of form `require('lazy-cache')(require)`.
+   */
+  class LazyCacheObject extends DataFlow::SourceNode {
+    LazyCacheObject() {
+      // Use `require` directly instead of `moduleImport` to avoid recursion.
+      // For the same reason, avoid `Import.getImportedPath`.
+      exists(Require req |
+        req.getArgument(0).getStringValue() = "lazy-cache" and
```

Comment:
Do we really want to mention `require` in the `LazyCacheObject` charpred? Or is it not worth generalizing to support other wrapped values?

Reply:
As the comment says, I don't want to introduce recursion because of this library. I'm also not sure it works with anything other than `require`.

Reply:
Doh. I had an off by one here. I thought you were hard-coding the `lazy-cache` model to only support wrapping of `require`. My bad.